master/buildbot/worker/ec2.py

```diff
@@ -71,7 +73,7 @@ def __init__(self, name, password, instance_type, ami=None,
                 "Use of default value of 'keypair_name' of EC2LatentWorker "
                 "constructor is deprecated. Please explicitly specify value")
             keypair_name = 'latent_buildbot_slave'
-        if security_name is None:
+        if security_name is None and not subnet_id:
             reportDeprecatedWorkerNameUsage(
```

Comment:
@sa2ajj during transition to worker terminology default `security_name` was changed from `'latent_buildbot_slave'` to "unspecified".
Helper function is used for easier managing of worker-related deprecations --- I hope we will remove them all at once at some point, plus it's not just `DeprecationWarning`, it's `DeprecatedWorkerNameWarning`.

@ryansydnor what would you do if `EC2LatentWorker.__init__` would have `security_name` keyword argument defaulted to `'latent_buildbot_slave'`?

```
class EC2LatentWorker(AbstractLatentWorker):
    def __init__(self, ..., security_name='latent_buildbot_slave', ...):
        ...
```

You would make it defaulted to `None` and check which one of `security_name` or `subnet_id` is specified?
Am I correct that `security_name` and `subnet_id` are exclusive options and one and only one option can be used?